scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py

```diff
@@ -384,6 +329,27 @@ def get_last_added(self):
     def get_sample_log_from_ws(self, ws_name, log_name):
         return self._loaded_workspaces[ws_name].getSampleDetails().getLogData(log_name).value
 
+    def _convert_TOF_to_d(self, tof, ws_name):
+        difa, difc, tzero = self._get_diff_constants(ws_name)
+        if abs(difa) < 1E-10:
+            return (tof - tzero) / difc
+        else:
+            return (-difc + sqrt(difc ** 2 - 4 * difa * (tzero - tof))) / (2 * difa)
```

Comment:
Hmm, I'm not feeling 100% about us hard-coding this conversion to d-spacing.

Do you know if Mantid has some sort of algorithm / method which could do this for us, so that we have one place where we do conversions rather than it implements in `n` places leading to potentially `n` bug fixes?

Reply:
To clarify, I know that `ConvertUnits` won't work, as this is a single value rather than a workspace, but I can't think if anything exists for converting single values (though it should)

Reply:
I definitely agree with your comment, since merging this PR #30163 I believe there is such functionality for unit conversion exposed to python (see Framework/PythonInterface/test/python/mantid/kernel/UnitConversionTest.py for an example) - this PR was intended to be merged before #30163 but it seems to have been neglected so now I can take advantage of this new stuff.

Reply:
I have changed this to use the UnitConverter introduced in #31063